### PR TITLE
Handle error in downloading coverage data

### PR DESCRIPTION
**Pull request by Navidem**
Had to return empty list if download_coverage_data_steps fails.

### CODE REVIEW

**Comment by Navidem** [comment]
It is better to check for errors in the caller of `download_coverage_data_steps`.

File: infra/build/functions/build_and_run_coverage.py [hunk 1]
```diff
@@ -295,11 +295,13 @@ def get_fuzz_introspector_steps(  # pylint: disable=too-many-locals, too-many-ar
...
-  build_steps.extend(
-      build_lib.download_coverage_data_steps(project.name,
-                                             coverage_report_latest,
-                                             bucket_name, build.out,
-                                             config.testing))
+  download_coverage_steps = build_lib.download_coverage_data_steps(
+      project.name, coverage_report_latest, bucket_name, build.out,
+      config.testing)
+  if not download_coverage_steps:
+    logging.info('Skipping introspector build for %s.', project.name)
```
Comment (oliverchang) [inline infra/build/functions/build_and_run_coverage.py:302]: We should be more explicit here about why we're skipping this. i.e. no coverage data found. Also use `logging.warning` instead here.

**Review by oliverchang — APPROVED** [review]
LGTM with a comment

File: infra/build/functions/build_and_run_coverage.py (hunk 1 above)
Comment (Navidem) [reply infra/build/functions/build_and_run_coverage.py]: Applied your comment.

**Pull request merged**